protocols/luci-proto-3g/htdocs/luci-static/resources/protocol/3g.js

```diff
@@ -26,13 +26,13 @@ function write_keepalive(section_id, value) {
 	    f = (f_opt != null) ? +f_opt[0].formvalue(section_id) : null,
 	    i = (i_opt != null) ? +i_opt[0].formvalue(section_id) : null;
 
-	if (f == null || f == '' || isNaN(f))
-		f = 0;
+	if (f === '' || isNaN(f))
+		f = null;
 
 	if (i == null || i == '' || isNaN(i) || i < 1)
```

Comment:
Can't interval be zero also? I don't know for sure. I get the impression that setting it to zero should turn it off. Does that happen in the init scripts?

https://github.com/ppp-project/ppp/blob/616102e93be1c629821cafd83a89d9e6c0467033/pppd/lcp.c#L2492-L2493

Reply:
As long as failure is less than zero it should work.

https://github.com/openwrt/openwrt/blob/6720c4ccba256186bf2f1b1edadb851c447e62a5/package/network/services/ppp/files/ppp.sh#L133

https://github.com/openwrt/openwrt/blob/6720c4ccba256186bf2f1b1edadb851c447e62a5/package/network/services/ppp/files/ppp.sh#L145

Reply:
> As long as failure is less than zero it should work.
> 
> https://github.com/openwrt/openwrt/blob/6720c4ccba256186bf2f1b1edadb851c447e62a5/package/network/services/ppp/files/ppp.sh#L133

This just means that setting it to zero disables it (in the init scipt) - which is inaccessible with the`i < 1` behaviour. I'm not nit-picking your code- just what's going on here in general. That whole logic could do with a rewrite. Are you game?

Reply:
It can't be less than 1 on luci since there's a check for that
https://github.com/openwrt/luci/blob/b9496f2023cb6b5f51dff88f1cf3a3063390dae4/protocols/luci-proto-3g/htdocs/luci-static/resources/protocol/3g.js#L145

But if we want users to be able to disable keepalive through the interface by either setting the failure or/and the interval to 0 then yes we'd need to edit the init script too.

I think the init script should be edited regardless to always set the keepalive instead of omitting it when it's 0 since that way pppd may use whatever it's set to on options file.
https://github.com/openwrt/openwrt/blob/6720c4ccba256186bf2f1b1edadb851c447e62a5/package/network/services/ppp/files/etc/ppp/options